4-js/index.js

```diff
@@ -0,0 +1,330 @@
+  if (check !== 'dont-delete') {
+    tableUser.innerHTML = '';
+  }
+  for (let i = 0; i < arr.length; i++) {
+    const tr = document.createElement('tr');
+    tr.id = arr[i].id;
+    tr.classList.add('row');
+
+    const tdCheckbox = document.createElement('td');
+    const checkbox = document.createElement('input');
+    const label = document.createElement('label');
+    const span = document.createElement('span');
+    checkbox.classList.add('custom-input');
+    checkbox.classList.add('user-checkbox');
+    checkbox.name = arr[i].id;
+    span.classList.add('custom-span');
+    label.classList.add('custom-checkbox');
+    checkbox.type = 'checkbox';
+    label.appendChild(checkbox);
+    label.appendChild(span);
+    tdCheckbox.appendChild(label);
+    tr.appendChild(tdCheckbox);
+
+    createTdElement(arr[i].company, tr);
+    createTdElement(arr[i].name, tr);
+    createTdElement(arr[i].address, tr);
+    createTdElement(arr[i].city, tr);
+    createTdElement(arr[i].country, tr);
+    createTdButtonElement(arr[i].id, tr, 'btn-edit');
+    createTdButtonElement(arr[i].id, tr, 'btn-delete');
+
+    tableUser.insertBefore(tr, tableUser.firstChild);
+  }
+}
+
+function createTdElement(value, parent) {
+  const td = document.createElement('td');
+  td.innerText = value;
+  parent.appendChild(td);
+}
+
+function createTdButtonElement(value, parent, style) {
+  const button = document.createElement('button');
+  const td = document.createElement('td');
+  button.name = value;
+  button.classList.add(style);
+  td.appendChild(button);
+  parent.appendChild(td);
+  if (style === 'btn-edit') {
+    button.addEventListener('click', showUserForm);
+  } else {
+    button.addEventListener('click', checkCheckbox);
+  }
+}
+
+function checkCheckbox(el) {
+  if (
+    el.currentTarget.parentNode.parentNode.children[0].children[0].children[0]
+      .checked
+  ) {
+    deleteUser(
+      el.currentTarget.parentNode.parentNode.children[0].children[0].children[0]
+    );
+  } else {
+    alert('The checkbox should be highlighted!');
+  }
+}
+
+function deleteUser(el) {
+  sendRequsest('DELETE', requestUrl + '/' + el.name)
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+  el.parentNode.parentNode.parentNode.remove();
+}
+
+function showUserForm(el) {
+  formUser.classList.remove('is-hide');
+  editUserBtn.classList.remove('is-hide');
+  titleEditUser.classList.remove('is-hide');
+  getEditObj(el.currentTarget.name);
+}
+
+function getEditObj(id) {
+    console.log(id);
+  sendRequsest('GET', requestUrl + '/' + id)
+    .then((data) => addValueToForm(data))
+    .catch((err) => console.log(err));
+}
+
+function addValueToForm(obj) {
+  userCompany.value = obj.company;
+  userCompany.name = obj.id;
+  userName.value = obj.name;
+  userAddress.value = obj.address;
+  userCity.value = obj.city;
+  userCountry.value = obj.country;
+}
+
+function editUser() {
+  let newUser = {
```

Comment:
the same as for createUser, I would pass an object of new user as a payload

Reply:
done